emcc.py

```diff
@@ -2394,7 +2386,7 @@ def generate_html(target, options, js_target, target_basename,
 %s
           };
           emterpretXHR.send(null);
-''' % (shared.Settings.EMTERPRETIFY_FILE, script.inline)
+''' % (shared.JS.get_subresource_location(shared.Settings.EMTERPRETIFY_FILE), script.inline)
```

Comment:
Yep, requesting a data URI through XHR/fetch works as expected; it's just in the node and shell environments where it wouldn't work since they would try to read it from the filesystem.

The drawback to relying on XHR for this in a web page is just the CSP thing — that a `Content-Security-Policy` header would need to whitelist `data:` — which I documented as a limitation of combining `SINGLE_FILE` with HTML output. This would eventually fix itself when/if it's refactored to use the same implementations from shell.js, and until then probably wouldn't be a common issue.

As a demonstration, you can compare running `(async () => console.log(await (await fetch('data:text/plain;base64,YmFsbHMK')).text()))()` in a JS console on GitHub (which has a strict CSP) and almost any other site.